sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -390,8 +390,20 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(OSModuleHandle M,
         !SYCLConfig<SYCL_DEVICELIB_NO_FALLBACK>::get())
       DeviceLibReqMask = getDeviceLibReqMask(Img);
 
+    std::string CompileOpts = Img.getCompileOptions();
+    if (Img.getSyclIsEsimdImage().isAvailable()) {
```

Comment:
NIT: this code seems quite verbose. Could be simplified to something like:
```
if (Img.isBoolPropertyTrue(/*category*/PropertySetRegistry::SYCL_IS_ESIMD_IMAGE, /*property name*/"IsEsimdImage")
{...}
```

Reply:
Fixed.

Reply:
I added a new method `isBoolPropertyTrue` to the `DeviceBinaryImage` class.